_posts/2019-11-01-deploying-debezium-with-kafkaconnector-resource.md

```diff
@@ -0,0 +1,697 @@
+* I'm using `database.hostname: 192.168.99.1` as IP address for connecting to MySQL because I'm using `minikube` with the virtualbox VM driver
+  If you're using a different VM drvier with `minikube` you might need a different IP address.
+* The `database.port: "3306"` works because of the `-p 3306:3306` argument we used when we started up the MySQL server.
+* The `database.whitelist: "inventory"` basically tells Debezium to only watch the `inventory` database.
+* The `database.history.kafka.topic: "schema-changes.inventory"` configured Debezium to use the `schema-changes.inventory` topic to store the database schema history.
+
+A while after you've created this connector you can have a look at its `status`, using `kubectl get kctr inventory-connector -o yaml`:
+
+```
+...
+status:
+  conditions:
+  - lastTransitionTime: "2019-11-18T15:01:53.135Z"
+    status: "True"
+    type: Ready
+  observedGeneration: 1
+```
+
+This tells us that the connector is running within the `KafkaConnect` cluster we created in the last step.
+
+
+To summarise, we've now the complete picture with the connector talking to MySQL:
+
+<pre style="line-height: 1.2 !important;"><code>
+┌────────────────────────────────┐         ┌────────────────────────────────┐
+│ minikube, namespace: kafka     │         │ docker                         │
+│                                │         │                                │
+│ Kafka                          │     ┏━━━┿━▶ MySQL                        │
+│  name: my-cluster              │     ┃   │                                │
+│                                │     ┃   └────────────────────────────────┘
+│ KafkaConnect                   │     ┃
+│  name: my-connect              │     ┃
+│                                │     ┃
+│ KafkaConnector                 │     ┃
+│  name: inventory-connector ◀━━━┿━━━━━┛
+│                                │
+└────────────────────────────────┘
+</code></pre>
+
+> OK, I admit it: I was lying about the ASCII art being awesome.
+
+
+# Showtime!
+
+If you list the topics, for example using `kubectl exec my-cluster-kafka-0 -c kafka -i -t -- bin/kafka-topics.sh --zookeeper '' --list` you should see:
+
+```
+__consumer_offsets
+connect-cluster-configs
+connect-cluster-offsets
+connect-cluster-status
+dbserver1
+dbserver1.inventory.addresses
+dbserver1.inventory.customers
+dbserver1.inventory.geom
+dbserver1.inventory.orders
+dbserver1.inventory.products
+dbserver1.inventory.products_on_hand
+schema-changes.inventory
+```
+
+The `connect-cluster-*` topics are the usual internal Kafka Connect topics.
+Debezium has created a topic for the server itself (`dbserver1`), and one for each table within the `inventory` database.
+
+Let's start consuming from one of those change topics:
+
+```
+kubectl exec my-cluster-kafka-0 -c kafka -i -t -- \
```

Comment:
For Debezium examples we moved to kafkacat lately, as it's much faster to start and also allows for nicer formatting/filtering via jq. E.g. see here if you're interested: https://github.com/debezium/debezium-examples/tree/master/auditlog#modifying-some-data-and-observing-the-audit-log

We provide a container image, which contains these tools, e.g. also mycli and pgcli as nicer CLI clients for the DBs. Just some food for thought.

Reply:
Strimzi's existing images don't include kafkacat. I not keen to distract from the main thrust of the content by having to deploy some other pod just to show the records in the topics in a marginally better way. Maybe exposing the cluster and accessing from outside would be less distracting.

But I take your point (and it applies not just to this blog, but to other blogs and to demos) that it's a bit awkward showing messages in topics with Strimzi at the moment. The whole `exec`ing in the broker pod is ugly and it misleads people into thinking that this is the correct way to do it.

Reply:
Sure. I guess it's a bit more difficult to do indeed with K8s where a pod needs to be deployed. In our READMEs we simply use `docker run` with out tooling image and then kafkacat etc., so it doesn't add any overhead in terms of perceived complexitiy.